### PR TITLE
Implement set_alpha_channel in Colorbar

### PR DESCRIPTION
**Pull request by ameraner**
This PR adds the possibility to define an alpha channel range to be applied to a colormap.

 - [x] Tests added (for all bug fixes or enhancements)
 - [x] Tests passed (for all non-documentation changes)
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` (remove if you did not edit any Python files)
 - [x] Fully documented (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

File: trollimage/colormap.py [hunk 1]
```diff
@@ -365,6 +365,41 @@ def set_range(self, min_val, max_val, inplace=True):
...
+            inplace (bool): If True (default), modify the values inplace.
+                If False, return a new Colormap instance.
+
+        """
+        if inplace:
+            cmap = self
+        else:
+            cmap = Colormap(
+                values=self.values.copy(),
+                colors=self.colors.copy())
```
Comment (djhoese) [inline trollimage/colormap.py:390]: I'm pretty surprised we don't just have a `.copy` method...even if just internal (`._copy`) but so be it. Just wanted to mention it.

File: trollimage/colormap.py [hunk 2]
```diff
@@ -365,6 +365,41 @@ def set_range(self, min_val, max_val, inplace=True):
...
+
+        alpha = np.linspace(min_alpha,
+                            max_alpha,
+                            self.colors.shape[0])
```
Comment (djhoese) [inline trollimage/colormap.py:394]: If min and max alpha are the same value (ex. all 1s) does this still work?

File: trollimage/colormap.py (hunk 1 above)
Comment (ameraner) [reply trollimage/colormap.py:390]: Yeah, I was surprised as well

File: trollimage/colormap.py (hunk 2 above)
Comment (ameraner) [reply trollimage/colormap.py:394]: Yes, the linspace just replicates the number and the alpha is then constant across the colorbar, just tested this also in the wild just to be sure.
```
np.linspace(1,1,10)
Out[14]: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```

**Comment by djhoese** [comment]
I'm OK with this being merged, but I guess I'm confused at how this hasn't come up before. Is there some other way to accomplish this besides making a custom colormap each time?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -59,7 +59,6 @@ jobs:
           git+https://github.com/dask/distributed \
           git+https://github.com/rasterio/rasterio \
           git+https://github.com/pydata/bottleneck \
-          git+https://github.com/zarr-developers/zarr \
```
Comment (djhoese) [inline .github/workflows/ci.yaml:62]: The newest dev version of zarr was causing the unstable environment to fail at install. I made a similar change in Satpy's CI.

**Comment by djhoese** [comment]
So CI is failing now but I don't think it is from my change (I didn't change the stable/regular environments), but rather than numpy 2.0 is now available on conda-forge. Looks like the conda action no longer shows the version of packages used/installed by default or at least I'm not seeing it in the log. It is likely that the recent xarray changes (and probably release) are breaking things from what we expected. We will probably see this in Satpy too as xarray's fixes for casting rule changes in numpy required changes in xarray and they don't seem to be producing exactly the same results.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (ameraner) [reply .github/workflows/ci.yaml:62]: Thanks!

**Comment by ameraner** [comment]
> I'm OK with this being merged, but I guess I'm confused at how this hasn't come up before. Is there some other way to accomplish this besides making a custom colormap each time?

Not that I could tell, no. I guess if you stay natively in trollimage to generate a colormap, it's easy enough to manipulate the alpha channel manually. And I guess that from the Satpy side people usually need fully opaque colorbars for overlays, or use the `MaskingCompositor` to set hard transparency transitions. There's also the `CloudCompositor` which manipulates the alpha channel directly at RGBA composite level, rather than at colorbar level.

> So CI is failing now but I don't think it is from my change (I didn't change the stable/regular environments), but rather than numpy 2.0 is now available on conda-forge. Looks like the conda action no longer shows the version of packages used/installed by default or at least I'm not seeing it in the log. It is likely that the recent xarray changes (and probably release) are breaking things from what we expected. We will probably see this in Satpy too as xarray's fixes for casting rule changes in numpy required changes in xarray and they don't seem to be producing exactly the same results.

Yeah, looks like it :/ Right now I don't have time to look into that unfortunately.

**Comment by djhoese** [comment]
I merged `main` into this PR to fix CI. I was about to merge it, but realized I'm the only one who has actually reviewed it. I'd like @mraspaud or maybe @pnuu to review it just to make sure there isn't an obvious better approach to this functionality.

**Comment by djhoese** [comment]
And if you merge it, feel free to make a minor release. If you don't I can when I start work, but it might save Andrea time to have this merged and released so he can work on his Satpy PR.

**Review by pnuu — APPROVED** [review]

**Review by mraspaud — APPROVED** [review]
LGTM!

**Pull request merged**